gke-whereami/app.py

```diff
@@ -0,0 +1,150 @@
+import requests
+from flask import Flask, request, Response, jsonify
+import logging
+import json
+import sys
+import socket
+import os
+from datetime import datetime
+import emoji
+import random
+from flask_cors import CORS
+
+METADATA_URL = 'http://metadata.google.internal/computeMetadata/v1/'
+METADATA_HEADERS = {'Metadata-Flavor': 'Google'}
+
+app = Flask(__name__)
+app.config['JSON_AS_ASCII'] = False  # otherwise our emojis get hosed
+CORS(app)  # enable CORS
+
+# set up emoji list
+emoji_list = list(emoji.unicode_codes.UNICODE_EMOJI.keys())
+
+
+@app.route('/')
+def home():
+
+    # get project ID
+    try:
```

Comment:
As we discussed, in each try block, check for null value of `os.getenv` and omit the key from the JSON response if it does not exist. The goal is to be able to do this and have the app be fully functioning for the simplest possible demo:

```
kubectl run --image=gcr.io/alexmattson-scratch/whereami --labels app=whereami --expose --port 8080
```

This should still output information that will always be in the pod like timestamp and hostname. If the user wants to do a more complex demo they can simply add just the environment variables they want to the ConfigMap.

Reply:
done